src/base/relation.js

```diff
@@ -14,7 +14,7 @@ export default class RelationBase {
       this.targetTableName   = result(Target.prototype, 'tableName');
       this.targetIdAttribute = result(Target.prototype, 'idAttribute');
     }
-    assign(this, { type, target: Target }, options);
+    assign(this, { type, target: Target }, omitBy(options, isNil));
```

Comment:
@ricardogama, if done how I [commented](https://github.com/tgriesser/bookshelf/pull/1397#issuecomment-250721474), you would need this change.